R/run_stress_test.R

```diff
@@ -1,55 +1,126 @@
 #' Run stress testing for provided asset type.
 #'
-#' @param asset_type String holding asset_type, for allowed value compare
+#' This function runs the transition risk stress test. It can be desirable to
+#' understand sensitivities of the scenarios, in which case the user may pass a
+#' vector of values to one (and only one) of the detail arguments. This will
+#' result in running the analysis multiple times in a row with the argument
+#' varied.
+#' NOTE: argument `asset_type_arg` is not iterateable.
```

Comment:
FIXME: My experiment suggests otherwise:

``` r
devtools::load_all(".")
#> ℹ Loading r2dii.climate.stress.test

# Should fail before any computation but doesn't
forbiden <- c("bonds", "loans")
run_stress_test(asset_type_arg = forbiden)
#> Running transition risk stress test 
#> Validating input arguments. 
#> -- Configuring analysis settings. 
#> -- Importing and preparing input data from designated input path. 
#>       >> When joining PACTA results on financial data on column(s) company_name, corporate_bond_ticker dropped rows for 36 out of 193 companies
#>         >> percent loss: 18.65285 
#>         >> affected companies:
#>           >> Abu Dhabi National Energ 
#>           >> Baltimore Gas & Electric Co 
#>           >> Bmw Intl Investment Bv 
#>           >> Cnooc Curtis Funding No 1 Pty Ltd 
#>           >> Cnooc Finance 2015 Au 
#>           >> Drax Finco Plc 
#>           >> Ebn Bv 
#>           >> Emera Us Finance Lp 
#>           >> Emgesa Sa Esp 
#>           >> Empresa Nacional Del Pet 
#>           >> Enbw Intl Finance Bv 
#>           >> Enel Finance Intl Nv 
#>           >> Entergy Louisiana Llc 
#>           >> Exxon Mobil Corporation 
#>           >> Fortis Inc 
#>           >> Frontera Energy Corp 
#>           >> Georgia Power Company 
#>           >> Hess Corp 
#>           >> Interstate Power & Light Co 
#>           >> Kondor Finance Plc (Nak) 
#>           >> Lg&E & Ku Energy Llc 
#>           >> Occidental Petroleum Cor 
#>           >> Perusahaan Listrik Negara Pt 
#>           >> Public Service Electric & Gas 
#>           >> Rci Banque Sa 
#>           >> Scana Corporation 
#>           >> Sinopec Grp Oversea 2012 
#>           >> Sinopec Grp Oversea 2013 
#>           >> Sinopec Grp Oversea 2014 
#>           >> Sinopec Grp Oversea 2015 
#>           >> Total Capital Intl Sa 
#>           >> Total Sa 
#>           >> Tullow Oil Plc 
#>           >> Vistra Operations Co Llc 
#>           >> Volkswagen Leasing Gmbh 
#>           >> Yankuang Group Cayman Ltd 
#> -- Calculating market risk.
#> Warning: Removed 21 rows because negative production compensation targets were set in late and sudden.
#>                    Negative absolute production is impossible
#>       >> When filtering out holdings with exposures of 0 value, dropped rows for 5 out of 157 companies
#>         >> percent loss: 3.184713 
#>         >> affected companies:
#>           >> Clp Power Hk Financing 
#>           >> Glencore Canada Financial Corp 
#>           >> Ongc Videsh Ltd 
#>           >> Origin Energy Finance Ltd 
#>           >> Pt Adaro Indonesia 
#> -- Calculating credit risk. 
#> Validating input arguments. 
#> -- Configuring analysis settings. 
#> -- Importing and preparing input data from designated input path.
#> Error in read_pacta_results(path = path, level = calculation_level): valid_input_file_path is not TRUE
```

<sup>Created on 2021-11-09 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Reply:
fixed with https://github.com/2DegreesInvesting/r2dii.climate.stress.test/pull/175/commits/e3f95b97a9ed0039c5ac34d8f9974616f575be05

I hope this is not the only one I overlooked, it is clearly a good idea to get rid of the _arg sillyness and use your suggestion of only renaming when appending to results.